### PR TITLE
Added modal

### PR DESCRIPTION
**Pull request by SEGMK**
-added html modal element
-added modal logic
-added SiteHexGraber

### CODE REVIEW

**Comment by FoRoKo1o** [comment]
🚢

File: html/board.html [hunk 1]
```diff
@@ -16,7 +18,7 @@
                         <div class="mb-md-5 mt-md-4 pb-5">
                             <h2 class="fw-bold mb-2 text-uppercase">Main</h2>
                             <p class="text-white-50 mb-5 user-data">This is my user data</p>
-                            <div id="hexboard" class="hexboard">
+                            <div id="hexboard" class="hexboard"> <!--huuuuuuuuge pice of... board-->
```
Comment (FoRoKo1o) [inline html/board.html:21]: I dont like this comment :)

File: html/board.html [hunk 2]
```diff
@@ -149,15 +151,35 @@ <h2 class="fw-bold mb-2 text-uppercase">Main</h2>
                                 <div class="hexagon dissabled" id="130"></div>
                                 <div class="hexagon dissabled" id="131"></div>
                                 <div class="hexagon dissabled" id="132"></div>
-                            </div>
```
Comment (FoRoKo1o) [inline html/board.html:152]: was is das?

File: css/myModal.css [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  /* Modal Content/Box */
+  .myModal-content {
+    position: relative;
+    margin: auto;
+    border-radius: 1.5vh;
+    background-color: #212121;
+    margin: 5% auto; /* 15% from the top and centered */
+    padding: 20px;
+    border: 1px solid #888;
+    width: 50%; /* Could be more or less, depending on screen size */
```
Comment (FoRoKo1o) [inline css/myModal.css:25]: Very good comments from w3

File: css/myModal.css [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  }
+  
+  /* The Close Button */
```
Comment (FoRoKo1o) [inline css/myModal.css:28]: Do we use that?
Do we need to close modal when we promote pawn?
What if someone close modal without promotion?

File: js/promotionModalController.js
Comment (FoRoKo1o) [inline js/promotionModalController.js:1]: We need to test that

**Review by FoRoKo1o — CHANGES_REQUESTED** [review]
Close modal will brake our game.

**Review by FoRoKo1o — APPROVED** [review]

**Comment by SEGMK** [comment]
looks good (but with classes would look even better) 🚢

**Pull request merged**